sql/core/src/main/scala/org/apache/spark/sql/execution/SQLExecution.scala

```diff
@@ -128,7 +128,8 @@ object SQLExecution extends Logging {
           sparkPlanInfo = SparkPlanInfo.EMPTY,
           time = System.currentTimeMillis(),
           modifiedConfigs = redactedConfigs,
-          jobTags = sc.getJobTags()
+          jobTags = sc.getJobTags(),
+          jobGroupId = Some(sc.getLocalProperty(SparkContext.SPARK_JOB_GROUP_ID))
```

Comment:
This should probably be

```suggestion
          jobGroupId = Option(sc.getLocalProperty(SparkContext.SPARK_JOB_GROUP_ID))
```

so that we don't wind up with`Some(null)` : https://github.com/databricks/scala-style-guide?tab=readme-ov-file#option